project/SparkBuild.scala

```diff
@@ -873,10 +873,14 @@ object SparkConnectClient {
     },
 
     (assembly / assemblyShadeRules) := Seq(
-      ShadeRule.rename("io.grpc.**" -> "org.sparkproject.connect.client.grpc.@0").inAll,
-      ShadeRule.rename("com.google.protobuf.**" -> "org.sparkproject.connect.protobuf.@1").inAll,
-      ShadeRule.rename("com.google.common.**" -> "org.sparkproject.connect.client.guava.@1").inAll,
-      ShadeRule.rename("com.google.thirdparty.**" -> "org.sparkproject.connect.client.guava.@1").inAll,
+      ShadeRule.rename("io.grpc.**" -> "org.sparkproject.connect.client.io.grpc.@1").inAll,
+      ShadeRule.rename("com.google.**" -> "org.sparkproject.connect.client.com.google.@1").inAll,
+      ShadeRule.rename("io.netty.**" -> "org.sparkproject.connect.client.io.netty.@1").inAll,
```

Comment:
There are still some differences between the packaging results of maven-shade and sbt-assembly. For example, netty does not exist in `spark-connect-client-jvm-assembly-3.5.0-SNAPSHOT.jar`

Reply:
Fixed the netty issue. I still feel it is the best to do allowlist so that we do not miss anything. The maven cannot do allowlist because the shading rule defined in the parent pom.